### PR TITLE
Support org url param

### PR DESCRIPTION
**Pull request by wimagguc**
**User-Facing Changes**

When accessing a url with an `org` parameter, the logged in user may be signed in with the incorrect org for that resource. With this PR we'll be able to recognize this and help the user take the necessary next step:

249537633-8412854e-1ea5-4d1e-9e43-052b94ae2b66

**Description**

Under the hood this approach only _extends_ on existing code paths, to sneak in the required changes:
* Besides `currentUserRequired`, we also return `userSwitchRequired` inside `useInitialDeepLinkState`
* The existing `SignInFormModal` component handles the signout confirmation dialog if needed

### CODE REVIEW

File: packages/studio-base/src/components/SignInFormModal.tsx [hunk 1]
```diff
@@ -4,13 +4,34 @@
...
   const [open, setOpen] = useState(true);
+  const { signOut } = useCurrentUser();
+  const [confirm, confirmModal] = useConfirm();
+
+  useEffect(() => {
+    if (!userSwitchRequired) {
+      return;
+    }
+    void confirm({ title: "You need to switch users", ok: "Sign out" }).then((response) => {
+      if (response === "ok") {
```
Comment (defunctzombie) [inline packages/studio-base/src/components/SignInFormModal.tsx:26]: What happens if the response is not ok?

File: packages/studio-base/src/hooks/useInitialDeepLinkState.ts [hunk 2]
```diff
@@ -150,21 +151,32 @@ export function useInitialDeepLinkState(deepLinks: readonly string[]): {
...
-  useSyncSourceFromUrl(targetUrlState, currentUserRequiredParam);
-  useSyncLayoutFromUrl(targetUrlState, currentUserRequiredParam);
+
+  useSyncSourceFromUrl(targetUrlState, { currentUserRequired });
+  useSyncLayoutFromUrl(targetUrlState, { currentUserRequired });
   useSyncTimeFromUrl(targetUrlState);
-  return currentUserRequiredParam;
+
+  const { currentUser } = useCurrentUser();
+  const userSwitchRequired = useMemo(
```
Comment (defunctzombie) [inline packages/studio-base/src/hooks/useInitialDeepLinkState.ts:173]: I don't think you need a memo here - the memo check itself isn't gonna save you much over the boolean expression from what I can tell.

**Comment by defunctzombie** [comment]
This makes sense to me. I'd like to see it in action.

File: packages/studio-base/src/components/SignInFormModal.tsx (hunk 1 above)
Comment (wimagguc) [reply packages/studio-base/src/components/SignInFormModal.tsx:26]: The confirm dialog disappears, so the user is free to open another data source or do something else in there. (The dialog won't reappear unless the user tries to reopen the link.)

**Comment by defunctzombie** [comment]
@wimagguc lets go with `org` as the url parameter and see how we feel about that

**Review by defunctzombie — APPROVED** [review]

**Pull request merged**